storybookjs/storybook
# Add preact docs support

Closes #

Related issues: 
https://github.com/storybookjs/storybook/issues/14876
https://github.com/storybookjs/storybook/issues/19739

## What I did
The Preact framework and renderer in Storybook still does not have an out-of-box support for docs annotations although the current [feature matrix](https://storybook.js.org/docs/configure/frameworks-feature-support#community-frameworks) shows full docs support for Preact, I tried to tackle this issue by:

1. adding the same docs support for the Preact renderer as we have today for React (mostly copied the src/docs folder from 
2. adding a new docs preset for preact-webpack following the same pattern found in the react-webpack preset  (see framework-preset-preact-docs.ts)
3. adding missing TS support for the preact-webpack5/default-ts sandbox storybook template: although this template is available when creating a sandbox, the proper TS entries under renderers/preact/template/cli were missing, the task was always creating JS components
4. adding a test file for the framework-preset-preact-docs preset

I have limited knowledge on the Storybook monorepo ecosystem so I probably missed some pieces (especially the test cases) or did not think on all the implications but these changes should be a good starter to have the docs support be complete for Preact as well. 

## Checklist for Contributors

### Testing

#### The changes in this PR are covered in the following automated tests:

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. Run a sandbox for template: yarn task --task generate --template preact-webpack5/default-ts
2. Open Storybook in your browser: from sandbox/preact-webpack5-default-ts run yarn storybook
3. Click on on Button example and notice the available Description and Default column values in ArgsType table


### Documentation

      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
